addons/pos_restaurant/static/src/js/multiprint.js

```diff
@@ -51,41 +68,33 @@ const PosResMultiprintOrderline = (Orderline) => class PosResMultiprintOrderline
         if (!this.pos.config.iface_printers) {
             return;
         }
-        if (typeof this.mp_dirty === 'undefined') {
-            // mp dirty is true if this orderline has changed
-            // since the last kitchen print
-            // it's left undefined if the orderline does not
-            // need to be printed to a printer.
-
-            this.mp_dirty = this.printable() || undefined;
-        }
+        this.uuid = this.uuid || uuidv4();
+        // mp dirty is true if this orderline has changed since the last kitchen print
+        this.mp_dirty = false
         if (!this.mp_skip) {
             // mp_skip is true if the cashier want this orderline
             // not to be sent to the kitchen
             this.mp_skip  = false;
         }
     }
-    // can this orderline be potentially printed ?
-    printable() {
-        return this.pos.db.is_product_in_category(this.pos.printers_category_ids_set, this.get_product().id);
-    }
+    //@override
     init_from_JSON(json) {
         super.init_from_JSON(...arguments);
-        this.mp_dirty = json.mp_dirty;
-        this.mp_skip  = json.mp_skip;
+        if (this.pos.config.iface_printers) {
+            this.uuid = json.uuid;
+            this.mp_skip = json.mp_skip;
+        }
     }
+    //@override
     export_as_JSON() {
-        var json = super.export_as_JSON(...arguments);
-        json.mp_dirty = this.mp_dirty;
-        json.mp_skip  = this.mp_skip;
-        return json;
-    }
-    set_quantity(quantity) {
-        if (this.pos.config.iface_printers && quantity !== this.quantity && this.printable()) {
-            this.mp_dirty = true;
+        const json = super.export_as_JSON(...arguments);
```

Comment:
we are overriding the same models in floors.js and in multiprint.js as they are in the same module it doesn't really make sense.

So we should either put everything in a "models.js" file, or create separated files like "PosOrder.js", 'PosOrderLine.js', ....

Reply:
Blame Frederic from 2014 but yeah I do agree. We can put everything in one file as it's not that big atm